### ISSUE TITLE
Errors after Upgrading-pre-1.12-Android-projects

### ISSUE THREAD

**Issue by dpedrinha** [flutter/flutter#61494]
## Steps to Reproduce
After updating flutter my last build threw a warning that I should do this: [https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects](https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects)

I did the Full-Flutter app migration and after that I'm getting the following errors when compiling:

```
Launching lib\main.dart on Android SDK built for x86 in debug mode...
Running Gradle task 'assembleDebug'...

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     D:\github-repo\myapp\android\app\src\main\AndroidManifest.xml:40:13-43:19: AAPT: error: resource style/NormalTheme (aka my.domain.myapp:style/NormalTheme) not found.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 28s
The built failed likely due to AndroidX incompatibilities in a plugin. The tool is about to try using Jetfier to solve the incompatibility.
Building plugin cloud_firestore...
Running Gradle task 'assembleAarRelease'...
Exception: The plugin cloud_firestore could not be built due to the issue above.
```

I didn't have a styles.xml so I created one under android/app/values/strings.xml:
```
<?xml version="1.0" encoding="utf-8"?>
<resources>
    <!-- Theme applied to the Android Window while the process is starting -->
    <style name="LaunchTheme" parent="@android:style/Theme.Black.NoTitleBar">
        <!-- Show a splash screen on the activity. Automatically removed when
             Flutter draws its first frame -->
        <item name="android:windowBackground">@drawable/launch_background</item>
    </style>
    <!-- Theme applied to the Android Window as soon as the process has started.
         This theme determines the color of the Android Window while your
         Flutter UI initializes, as well as behind your Flutter UI while its
         running.

         This Theme is only used starting with V2 of Flutter's Android embedding. -->
    <style name="NormalTheme" parent="@android:style/Theme.Black.NoTitleBar">
        <item name="android:windowBackground">@android:color/white</item>
    </style>
</resources>
```

I also chose to delete the MainActivity.java/kt files.

And I changed my AndroidManifest.xml to:
```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="br.com.mydomain.myapp">

    <!-- io.flutter.app.FlutterApplication is an android.app.Application that
         calls FlutterMain.startInitialization(this); in its onCreate method.
         In most cases you can leave this as-is, but you if you want to provide
         additional functionality it is fine to subclass or reimplement
         FlutterApplication and put your custom class here. -->
	<uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.USE_FULL_SCREEN_INTENT" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
<!--    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>-->
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
<!--    <application-->
<!--        android:name="io.flutter.app.FlutterApplication"-->
<!--        android:label="@string/app_name"-->
<!--        android:icon="@mipmap/ic_launcher">-->
    <application
        android:label="@string/app_name"
        android:icon="@mipmap/ic_launcher">
        <activity
            android:name="io.flutter.embedding.android.FlutterActivity"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
            <!-- This keeps the window background of the activity showing
                 until Flutter renders its first frame. It can be removed if
                 there is no splash screen (such as the default splash screen
                 defined in @style/LaunchTheme). -->
<!--            <meta-data-->
<!--                android:name="io.flutter.app.android.SplashScreenUntilFirstFrame"-->
<!--                android:value="true" />-->
            <meta-data
                android:name="io.flutter.embedding.android.SplashScreenDrawable"
                android:resource="@drawable/launch_background" />
            <meta-data
                android:name="io.flutter.embedding.android.NormalTheme"
                android:resource="@style/NormalTheme"
                />
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
            <intent-filter>
                <action android:name="android.intent.action.VIEW"/>
                <category android:name="android.intent.category.DEFAULT"/>
                <category android:name="android.intent.category.BROWSABLE"/>
                <data android:host="myapp.page.link" android:scheme="https"/>
                <data android:host="myapp.page.link" android:scheme="http"/>
            </intent-filter>
        </activity>
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
		<receiver android:name="com.dexterous.flutterlocalnotifications.ScheduledNotificationBootReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED"></action>
            </intent-filter>
        </receiver>
        <receiver android:name="com.dexterous.flutterlocalnotifications.ScheduledNotificationReceiver" />
    </application>
</manifest>
```

I also didn't have drawables and mipmap as the migration tutorial assumes so I created a new flutter project with `flutter create trash` and copied the files to my project.

My Android Studio is also warning me about an IDE update 4.0.1 but I didn't do it because it says many of my flutter plugins are incompatible. So I'm on 3.6.3.

What am I missing for this migration?

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, 1.20.0-8.0.pre.84, on Microsoft Windows [Version 10.0.18363.900], locale pt-BR)

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[√] Android Studio (version 3.6)
[!] VS Code (version 1.46.0)
    X Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[√] Connected device (1 available)

! Doctor found issues in 1 category.

```

</details>

**Comment by TahaTesser** [comment]
Hi @dpedrinha 
Just tried to reproduce with exact same styles.xml as you have, no issues
Based on this, it cannot find it, its not an issue with Flutter or doc itself 

Your package is bit different than error message's package 

```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="br.com.mydomain.myapp">
```

```
     D:\github-repo\myapp\android\app\src\main\AndroidManifest.xml:40:13-43:19: AAPT: error: resource style/NormalTheme (aka my.domain.myapp:style/NormalTheme) not found.

```

Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by dpedrinha** [comment]
@TahaTesser Thank you for your response.
To be honest I don't know if I disagree or not, I really have no idea what the problem is. I agree that it's probably not an issue with flutter but it was caused by the migration tutorial for flutter, so I believe this problem is related to the new linking between flutter and android. Or maybe the tutorial is just missing something.
All I know is that I followed the migration tutorial and now I can't compile.
About the package differences it's just because I hid my real address when posting here. They are the same on the app. I'll edit it.
Maybe my app is so old that it's missing something?
Any help to what to look for or where is much appreciated.
